### PR TITLE
JSON endpoint in main module

### PR DESCRIPTION
**Pull request by malcomio**
Issue #2986913 by shashikant52004, malcomio: Serve configuration as JSON for apps to consume

replaces #11

### CODE REVIEW

File: spalp.routing.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+entity.node.appjson:
+  path: '/spalp/{app_id}/json'
+  defaults:
+    _title: 'App Data'
+    _controller: '\Drupal\spalp\Controller\AppLandingJsonController::data'
+  requirements:
+    _permission: 'access content'
+    node: \d+
```
Comment (malcomio) [inline spalp.routing.yml]: `node` here seems wrong

File: src/Controller/AppLandingJsonController.php [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+   *   The JSON output.
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+   */
+  public function data($app_id = '') {
+    if (empty($app_id)) {
+      throw new NotFoundHttpException();
+    }
+
+    $language = \Drupal::languageManager()->getCurrentLanguage()->getId();
```
Comment (dakkusingh) [inline src/Controller/AppLandingJsonController.php]: > \Drupal calls should be avoided in classes, use dependency injection instead

File: src/Service/Core.php [hunk 3]
```diff
@@ -39,7 +40,7 @@ class Core {
    *   Module Handler Interface.
    */
   public function __construct(LoggerChannelFactoryInterface $loggerFactory,
-                              ModuleHandlerInterface $moduleHandler) {
+    ModuleHandlerInterface $moduleHandler) {
```
Comment (dakkusingh) [inline src/Service/Core.php]: is this a personal tabbing pref or something relating to a coding standard?

Comment (malcomio) [reply src/Service/Core.php]: auto-formatting, probably

**Pull request merged**